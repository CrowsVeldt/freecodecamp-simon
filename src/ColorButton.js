import React, { Component } from 'react'
import styled from 'styled-components'

class ColorButton extends Component {
  render () {
    return (
      <Button {...this.props}>
        {/* look up "ref" for playing the audio */}
        <audio>
          <source src={this.props.source} />
        </audio>
      </Button>
    )
  }
}

const Button = styled.button`
  height: 350px;
  width: 350px;
  border: 10px solid black;
  position: absolute;
  z-index: 1;

  border-radius: ${
    props => props.green ? '100% 0 0 0'
      : props.red ? '0 100% 0 0'
      : props.blue ? '0 0 100% 0'
      : '0 0 0 100%'
  };

  top: ${
    props => props.green ? '0px'
      : props.red ? '0px'
      : props.blue ? '350px'
      : '350px'
   };

  left: ${
    props => props.green ? '0px'
      : props.red ? '350px'
      : props.blue ? '350px'
      : '0px'
  };

  background-color: ${
    props => props.green ? 'green'
      : props.red ? 'red'
      : props.blue ? 'blue'
      : 'yellow'
  };
`

export default ColorButton
