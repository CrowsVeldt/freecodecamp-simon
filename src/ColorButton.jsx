import React, { Component } from 'react'
import styled from 'styled-components'

class ColorButton extends Component {

    handleClick = () => {
      this.audioElement.play()
    }

  render () {
    return (
      <Button
        ref={this.props.buttonRef}
        onClick={this.handleClick}
        active
        {...this.props}
      >
        <audio ref={audio => this.audioElement = audio}>
          <source src={this.props.source} />
        </audio>
      </Button>
    )
  }
}

const Button = styled.button`
  height: 350px;
  width: 350px;
  border: 10px solid black;
  position: absolute;
  z-index: 1;

  border-radius: ${
    props => props.green ? '100% 0 0 0'
      : props.red ? '0 100% 0 0'
      : props.blue ? '0 0 100% 0'
      : '0 0 0 100%'
  };

  top: ${
    props => props.green ? '0px'
      : props.red ? '0px'
      : props.blue ? '350px'
      : '350px'
   };

  left: ${
    props => props.green ? '0px'
      : props.red ? '350px'
      : props.blue ? '350px'
      : '0px'
  };

  background-color: ${
    props => props.green ? 'RGB(0, 155, 0)'
      : props.red ? 'RGB(155, 0, 0)'
      : props.blue ? 'RGB(0, 0, 155)'
      : 'RGB(155, 155, 0)'
  };

  &:active {
    background-color: ${
      props => props.green ? 'RGB(0, 255, 0)'
        : props.red ? 'RGB(255, 0, 0)'
        : props.blue ? 'RGB(0, 0, 255)'
        : 'RGB(255, 255, 0)'
      };
  }
`

export default ColorButton
