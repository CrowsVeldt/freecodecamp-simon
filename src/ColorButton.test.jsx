import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ColorButton from './ColorButton'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<ColorButton />, div)
})

it('renders correctly', () => {
  const tree = renderer
    .create(<ColorButton topLeft />)
  expect(tree).toMatchSnapshot()
})
