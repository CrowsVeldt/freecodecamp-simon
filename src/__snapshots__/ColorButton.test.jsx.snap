// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="sc-bdVaJa lfqBkn"
  onClick={[Function]}
>
  <audio>
    <source
      src={undefined}
    />
  </audio>
</button>
`;
